x-pack/test_serverless/api_integration/test_suites/observability/apm_api_integration/common/apm_api_supertest.ts

```diff
@@ -0,0 +1,131 @@
+import { InheritedFtrProviderContext } from '../../../../services';
+
+export function createApmApiClient(st: supertest.SuperTest<supertest.Test>) {
+  return async <TEndpoint extends APIEndpoint>(
+    options: {
+      type?: 'form-data';
+      endpoint: TEndpoint;
+    } & APIClientRequestParamsOf<TEndpoint> & { params?: { query?: { _inspect?: boolean } } }
+  ): Promise<SupertestReturnType<TEndpoint>> => {
+    const { endpoint, type } = options;
+
+    const params = 'params' in options ? (options.params as Record<string, any>) : {};
+
+    const { method, pathname, version } = formatRequest(endpoint, params.path);
+    const url = format({ pathname, query: params?.query });
+
+    const headers: Record<string, string> = { 'kbn-xsrf': 'foo' };
+
+    if (version) {
+      headers['Elastic-Api-Version'] = version;
+    }
+
+    let res: request.Response;
+    if (type === 'form-data') {
+      const fields: Array<[string, any]> = Object.entries(params.body);
+      const formDataRequest = st[method](url)
+        .set(headers)
+        .set('Content-type', 'multipart/form-data');
+
+      for (const field of fields) {
+        formDataRequest.field(field[0], field[1]);
+      }
+
+      res = await formDataRequest;
+    } else if (params.body) {
+      res = await st[method](url).send(params.body).set(headers);
+    } else {
+      res = await st[method](url).set(headers);
+    }
+
+    // supertest doesn't throw on http errors
+    if (res?.status !== 200) {
+      throw new ApmApiError(res, endpoint);
+    }
+
+    return res;
+  };
+}
+
+type ApiErrorResponse = Omit<request.Response, 'body'> & {
+  body: {
+    statusCode: number;
+    error: string;
+    message: string;
+    attributes: object;
+  };
+};
+
+export type ApmApiSupertest = ReturnType<typeof createApmApiClient>;
+
+export class ApmApiError extends Error {
+  res: ApiErrorResponse;
+
+  constructor(res: request.Response, endpoint: string) {
+    super(
+      `Unhandled ApmApiError.
+Status: "${res.status}"
+Endpoint: "${endpoint}"
+Body: ${JSON.stringify(res.body)}`
+    );
+
+    this.res = res;
+  }
+}
+
+async function getApmApiClient({
+  kibanaServer,
+  username,
+}: {
+  kibanaServer: UrlObject;
+  username: ApmUsername | 'elastic';
+}) {
+  const url = format({
+    ...kibanaServer,
+    auth: `${username}:${APM_TEST_PASSWORD}`,
+  });
+
+  return createApmApiClient(supertest(url));
+}
+
+export interface SupertestReturnType<TEndpoint extends APIEndpoint> {
+  status: number;
+  body: APIReturnType<TEndpoint>;
+}
+
+type ApmApiClientKey = 'slsUser';
+export type ApmApiClient = Record<ApmApiClientKey, Awaited<ReturnType<typeof getApmApiClient>>>;
+
+export async function getApmTestService({
+  getService,
+}: InheritedFtrProviderContext): Promise<ApmApiClient> {
+  const svlSharedConfig = getService('config');
+  const kibanaServer = svlSharedConfig.get('servers.kibana');
+
+  return {
+    slsUser: await getApmApiClient({
```

Comment:
What does the `sls` stand for?

Reply:
serverless.
source - sls is used as a acronym by the actual serverless platform on npm

Reply:
Do you believe it adds value to include the acronym, considering that we are using it in the serverless testing suite?

Reply:
Yes, Users in serverless realm are quite different from stateful realm. At least the way they are created and assigned roles. This means it could be quite different from regular Editor/Viewer users.

Reply:
> serverless. source - sls is used as a acronym by the actual serverless platform on npm

@achyutjhunjhunwala can you please share where they're using this acronym

Reply:
Sure, the serverless on npm uses `sls` as the acronym for their CLI
https://www.npmjs.com/package/serverless#:~:text=Serverless%20Framework%20commands.-,Deploying,-If%20you%20haven%27t